fhnw-ph-schreibberatung.csl

```diff
@@ -0,0 +1,476 @@
+          </if>
+          <else-if type="broadcast">
+            <text variable="container-title" font-style="normal" suffix="."/>
+          </else-if>
+        </choose>
+      </if>
+    </choose>
+  </macro>
+
+  <macro name="editor">
+    <names variable="editor">
+      <name name-as-sort-order="all" and="text" sort-separator=", " delimiter="; " delimiter-precedes-last="never"/>
+      <label form="short" prefix=" (" suffix=".)" strip-periods="true"/>
+    </names>
+  </macro>
+
+  <macro name="translator">
+    <names variable="translator">
+      <name name-as-sort-order="all" and="text" sort-separator=", " delimiter="; " delimiter-precedes-last="never"/>
+      <label form="verb" prefix=", " suffix="." strip-periods="true"/>
+    </names>
+  </macro>
+
+  <macro name="recipient">
+    <choose>
+      <if type="personal_communication">
+        <choose>
+          <if variable="genre">
+            <text variable="genre" text-case="capitalize-first"/>
+          </if>
+          <else>
+            <text term="letter" text-case="capitalize-first"/>
+          </else>
+        </choose>
+      </if>
+    </choose>
+    <names variable="recipient" delimiter=", ">
+      <label form="verb" prefix=" " text-case="lowercase" suffix=" "/>
+      <name and="text" delimiter=", "/>
+    </names>
+  </macro>
+
+  <macro name="contributors">
+    <names variable="author">
+      <name and="text" name-as-sort-order="all" sort-separator=", " delimiter="; " delimiter-precedes-last="never"/>
+      <label form="verb-short" prefix=" " suffix="." text-case="lowercase" strip-periods="true"/>
+      <substitute>
+        <text macro="editor"/>
+        <text macro="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+    <text macro="recipient"/>
+  </macro>
+
+  <macro name="contributors-short">
+    <names variable="author">
+      <name form="short" delimiter="/"/>
+      <substitute>
+        <names variable="editor"/>
+        <names variable="translator"/>
+      </substitute>
+    </names>
+    <text macro="anon"/>
+  </macro>
+
+  <macro name="interviewer">
+    <names variable="interviewer" delimiter=", ">
+      <label form="verb" prefix=" " text-case="capitalize-first" suffix=" "/>
+      <name and="text" delimiter=", "/>
+    </names>
+  </macro>
+
+  <macro name="archive">
+    <group delimiter=". ">
+      <text variable="archive_location" text-case="capitalize-first"/>
+      <text variable="archive"/>
+      <text variable="archive-place"/>
+    </group>
+  </macro>
+
+  <macro name="access">
+    <group delimiter=". ">
+      <choose>
+        <if type="graphic report" match="any">
+          <text macro="archive"/>
+        </if>
+        <else-if type="bill book graphic legal_case motion_picture report song article-journal article-magazine article-newspaper thesis chapter paper-conference" match="none">
+          <text macro="archive"/>
+        </else-if>
+      </choose>
```

Comment:
combine these two into a single if (i.e. just remove graphic and report from l. 172 and then use that as the if condition in l. 169